### PR TITLE
cirrus-ci conditional tasks

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
This PR adds `only_if` conditional task execution hooks to `cirrus-ci` to allow developers to easily disable one or more `cirrus-ci` tasks, as a convenience, during development.

For further details see [Cirrus-CI Conditional Task Execution](https://cirrus-ci.org/guide/writing-tasks/#conditional-task-execution).

Also, see [here](https://bjlittle-iris.readthedocs.io/en/latest/developers_guide/contributing_ci_tests.html#skipping-cirrus-ci-tasks) to view the rendered changes to the documentation from this PR.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -50,7 +60,7 @@ linux_task_template: &LINUX_TASK_TEMPLATE
     fingerprint_script:
       - wget --quiet https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
       - echo "${CIRRUS_OS} $(sha256sum miniconda.sh)"
-      - echo "$(date +%Y).$(($(date +%U) / ${CACHE_PERIOD})):${CONDA_CACHE_BUILD}"
+      - echo "$(date +%Y).$(($(echo $(date +%U) | sed 's/^0*//') / ${CACHE_PERIOD})):${CONDA_CACHE_BUILD}"
```
Comment (bjlittle) [inline .cirrus.yml:63]: This is interesting...

The `$(data +%U)` returns `week number of year, with Sunday as first day of week (00..53)`... as it happens, at this point of writing, this is `08`. However this is being interpreted by `bash` as an octal number, which is out of range :scream: ... so this is a fix. On my.

Comment (bjlittle) [reply .cirrus.yml:63]: Weirdly, as it turned out, it was easy for me to disable all `cirrus-ci` tasks but one, and test the fix without hogging resource unnecessarily.

Serendipity, is a wonderful thing...

Comment (bjlittle) [reply .cirrus.yml]: However `expr` is available... :+1:

File: docs/src/whatsnew/latest.rst [hunk 2]
```diff
@@ -124,6 +124,10 @@ This document explains the changes made to Iris for this release
 #. `@akuhnregnier`_ replaced `deprecated numpy 1.20 aliases for builtin types`_.
    (:pull:`3997`)
 
+#. `@bjlittle`_ added conditional task execution to `.cirrus.yml`_ to allow
+   developers to easily disable `cirrus-ci`_ tasks. See
+   :ref:`skipping Cirrus-CI tasks`. (:pull:`4018`)
```
Comment (tkknight) [inline docs/src/whatsnew/latest.rst:129]: Should link to pull 4019?

File: docs/src/developers_guide/contributing_ci_tests.rst [hunk 3]
```diff
@@ -18,29 +18,65 @@ Iris **master**.  The checks performed are:
...
+Skipping Cirrus-CI Tasks
+------------------------
+
+As a developer you may wish to not run all the CI tasks when you are actively
+developing e.g., you are writing documentation and there is no need for linting,
+or long running compute intensive testing tasks to be executed.
+
+As a convenience, it is possible to easily skip one or more tasks by setting
+the appropriate environment variable within the `.cirrus.yml`_ file to a
+non-empty string:
```
Comment (tkknight) [inline docs/src/developers_guide/contributing_ci_tests.rst:61]: May be useful to make **non-empty string** bold.

File: .cirrus.yml [hunk 4]
```diff
@@ -18,6 +18,16 @@ container:
 
 
 env:
+  # Skip specific tasks by name.
```
Comment (tkknight) [inline .cirrus.yml:21]: Would be useful to say something like this:

"Skip specific tasks by name.  Set to a non-empty string to skip."

**Review by tkknight — COMMENTED** [review]
I think this change will be super useful as it will reduce the time to wait for the CI tests to run.  A highly useful example would be to disable the docs until the final commit.  

Only a few minor points to consider in my review.

**Review by tkknight — APPROVED** [review]

**Pull request merged**